lib/components/ModalPortal.js

```diff
@@ -166,7 +168,7 @@ var ModalPortal = module.exports = React.createClass({
   },
 
   shouldBeClosed: function() {
-    return !this.props.isOpen && !this.state.beforeClose;
+    return !this.state.isOpen && !this.state.beforeClose;
```

Comment:
if the `Modal` is unmounting, it won't have a chance to update the `isOpen` prop to `false`; however, the internal `setState` call will update `state.isOpen` correctly.